This patch will update the Homebrew formula for Poppler to create libraries
suitable for building TeXworks for distribution.

It may break if the Hombrew tree drifts too far out of sync.

diff --git a/usr/local/Library/Formula/poppler.rb b/CMake/packaging/mac/poppler.rb
index 1238eae..84493b3 100644
--- a/usr/local/Library/Formula/poppler.rb
+++ b/CMake/packaging/mac/poppler.rb
@@ -1,3 +1,27 @@
+# This file contains a formula for installing Poppler on Mac OS X using the
+# Homebrew package manager:
+#
+#     http://mxcl.github.com/homebrew
+#
+# To install Poppler using this formula:
+#
+#     brew install --with-qt4 --enable-xpdf-headers path/to/this/poppler.rb
+#
+# Changes compared to Homebrew's standard Poppler formula:
+#
+#   - TeXworks-specific patches are applied to help Qt apps find the
+#     poppler-data directory.
+#
+#   - Poppler is patched to use native OS X libraries for font handling instead
+#     of Fontconfig. This removes any dependencies on the presence of X11
+#     fonts.
+#
+#   - Poppler is configured to use as few dependencies as possible. This
+#     reduces the number of dylibs that must be added to TeXworks.app when it
+#     is packaged for distribution.
+TEXWORKS_SOURCE_DIR = Pathname.new(__FILE__).realpath.dirname.join('../../..')
+TEXWORKS_PATCH_DIR = TEXWORKS_SOURCE_DIR + 'lib-patches'
+
 require 'formula'
 
 class PopplerData < Formula
@@ -14,7 +38,13 @@ class Poppler < Formula
   depends_on "qt" if ARGV.include? "--with-qt4"
 
   def patches
-    DATA
+    {
+      :p1 => [
+        #TEXWORKS_PATCH_DIR + 'poppler-qt4-globalparams.patch',
+        TEXWORKS_PATCH_DIR + 'poppler-mac-font-handling.patch',
+        TEXWORKS_PATCH_DIR + 'poppler-debug-mac-font-handling.patch'
+      ]
+    }
   end
 
   def options
@@ -25,17 +55,41 @@ class Poppler < Formula
   end
 
   def install
-    if ARGV.include? "--with-qt4"
-      ENV['POPPLER_QT4_CFLAGS'] = `pkg-config QtCore QtGui --libs`.chomp.strip
-      ENV.append 'LDFLAGS', "-Wl,-F#{HOMEBREW_PREFIX}/lib"
-    end
+    cmake_args = std_cmake_parameters.split
+
+    # Save time by not building tests
+    cmake_args.concat [
+      '-DBUID_CPP_TESTS=OFF',
+      '-DBUID_GTK_TESTS=OFF',
+      '-DBUID_QT3_TESTS=OFF',
+      '-DBUID_QT4_TESTS=OFF'
+    ]
 
-    args = ["--disable-dependency-tracking", "--prefix=#{prefix}"]
-    args << "--disable-poppler-qt4" unless ARGV.include? "--with-qt4"
-    args << "--enable-xpdf-headers" if ARGV.include? "--enable-xpdf-headers"
+    cmake_args << "-DWITH_Qt4=OFF" unless ARGV.include? "--with-qt4"
+    cmake_args << "-DENABLE_XPDF_HEADERS=ON" if ARGV.include? "--enable-xpdf-headers"
 
-    system "./configure", *args
-    system "make install"
+    # TeXworks-specific additions to minimize library dependencies
+    cmake_args.concat [
+      '-DENABLE_ABIWORD=OFF',
+      '-DENABLE_CPP=OFF',
+      '-DENABLE_LCMS=OFF',
+      '-DENABLE_LIBCURL=OFF',
+      '-DENABLE_LIBOPENJPEG=OFF',
+      '-DENABLE_SPLASH=ON', # Required
+      '-DENABLE_UTILS=OFF',
+      '-DENABLE_ZLIB=OFF',
+      '-DWITH_Cairo=OFF',
+      '-DWITH_Iconv=OFF',
+      '-DWITH_JPEG=OFF',
+      '-DWITH_PNG=OFF',
+      '-DWITH_Qt3=OFF',
+    ]
+
+    Dir.mkdir 'build'
+    Dir.chdir 'build' do
+      system 'cmake', '..', *cmake_args
+      system "make install"
+    end
 
     # Install poppler font data.
     PopplerData.new.brew do
@@ -43,20 +97,3 @@ class Poppler < Formula
     end
   end
 end
-
-__END__
-fix location of fontconfig:
-  http://www.mail-archive.com/poppler@lists.freedesktop.org/msg03837.html
-
---- a/cpp/Makefile.in	2010-07-08 20:57:56.000000000 +0200
-+++ b/cpp/Makefile.in	2010-08-06 11:11:27.000000000 +0200
-@@ -375,7 +375,8 @@
- INCLUDES = \
- 	-I$(top_srcdir)				\
- 	-I$(top_srcdir)/goo			\
--	-I$(top_srcdir)/poppler
-+	-I$(top_srcdir)/poppler \
-+	$(FONTCONFIG_CFLAGS)
- 
- SUBDIRS = . tests
- poppler_includedir = $(includedir)/poppler/cpp
